### PR TITLE
pom file update to build XOAI-5.0.0 #9339

### LINKED ISSUE #9339: Update Dataverse pom file to build with the 5.0.0 release of XOAI library
We have merged the PR #9310 without waiting for the official, final 5.0.0 release of XOAI. So our develop branch is currently being built with develop, 5.0.0-SNAPSHOT versions of it. Once the release is made, a quick PR will need to be made updating the version number in modules/dataverse-parent/pom.xml. So I'm opening this issue to keep track of that/make sure we remember to do this. 

It would be nice, to have this done before 5.13. But, thinking about it, it's NOT going to be a problem in any practical way, if we build and release from the develop branch as is, with the snapshots either. So this should not be considered a blocker or especially urgent.

On the XOAI side, this is tracked in https://github.com/gdcc/xoai/issues/59.

### PR DESCRIPTION
**Pull request by landreev**
**What this PR does / why we need it**:

As described in the linked issue, switching the application to the long-awaited final version of the XOAI v.5.0.0 that has just been released. 

**Which issue(s) this PR closes**:

Closes #9339

**Special notes for your reviewer**:

This uses the final, mature version of XOAI 5.0.0. It contains the latest changes, that were added to accommodate the PR #9310. Those were tested as part of the QA on that PR (built with the final release candidate version, essentially the same jars but packaged as dev. snapshots. The original plan was to wait for the 5.0.0 to be packaged before merging it, but then we merged it anyway; kind of by accident, but in retrospect, it would look bad for that PR to sit in QA all this time). 

The XOAI release in question is the mature version, and the culmination of all the work done to improve, update and modernize the library since it was forked off DSpace/xoai and moved into its new home at https://github.com/gdcc/xoai. Massive credit goes to @poikilotherm for making it happen and all the work he's done there. 

**Suggestions on how to test this**:
Functionality-wise, almost everything in this PR has gone through our QA process already. So it may only need a quick glance, confirming that nothing is visibly broken and the integration tests are passing. 
*The only small change/improvement you will notice* is the treatment of `yyyy-mm-dd` dates (the thing you noticed during QA of #9316)

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

**Additional documentation**:

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/harvest/server/web/servlet/OAIServlet.java [hunk 1]
```diff
@@ -205,7 +205,7 @@ private RepositoryConfiguration createRepositoryConfiguration() {
                 .withAdminEmail(systemEmailLabel)
                 .withCompression("gzip")
                 .withCompression("deflate")
-                .withGranularity(Granularity.Second)
+                .withGranularity(Granularity.Lenient)
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/harvest/server/web/servlet/OAIServlet.java:208]: What does this do? Was this change intentional?

**Review by pdurbin — COMMENTED** [review]
Just a quick question.

**Review by scolapasta — APPROVED** [review]

File: src/main/java/edu/harvard/iq/dataverse/harvest/server/web/servlet/OAIServlet.java (hunk 1 above)
Comment (landreev) [reply src/main/java/edu/harvard/iq/dataverse/harvest/server/web/servlet/OAIServlet.java:208]: It is the single actual change in the PR, it's mentioned under "how to test". 

It's to address a small issue @kcondon found the other day; not a huge thing, we were just a little too strict with parameter formatting and it was against the OAI spec.

Comment (pdurbin) [reply src/main/java/edu/harvard/iq/dataverse/harvest/server/web/servlet/OAIServlet.java:208]: Ok, the date thing. Thanks!

**Comment by poikilotherm** [comment]
Thanks for bearing with me! Glad we worked it all out.
